### DISCUSSION TITLE
Keep the file size after opening and saving without making any changes

### DISCUSSION THREAD

**Discussion by VadymIan** [dlemstra/Magick.NET#1476] [Help] [answered]
HI folks.
I have been trying to solve such an issue but still unsuccessful. Hope you can help me with it.
We are using magick.net to clean metadata in image files.

```
            using (var image = new MagickImage(imageStream))
            {
                    image.Settings.Compression = image.Compression;
                    image.Strip();
                    image.Write(outputName);
            }
```
this snipped is applied to different types of images, but it significantly reduces the size of the resulted file.
If I comment `image.Strip();` line and just open, apply initial compression and save it, it also reduces final size file.
According to reseach I have found possible solution to set up `Quality `parameter closer to 100. And it partially helps for jpg extensions but tif, png, jpg loses about 3 times it's size.

Any explanations and propositions are welcome.
Thank you!

**Comment by dlemstra** [comment]
I would advise you to use the `Compress` method of `ImageOptimizer` instead to reduce the file size of images. There is no support for tiff files so you will need to do that yourself. Do you have an example where the file size is increased? The answer depends on the image that are using.

**Comment by VadymIan** [comment]
But the point is that I don't want to compress file. 
I just want to clean metadata and resave it without reducing the size
`image.Settings.Compression = image.Compression;` this line of code is for saving initial compression for cleaned file
For example: there is a file: 1.00 MB (1,056,398 bytes).
after performing this code:
```
using (var image = new MagickImage(imageStream))
    {
            image.Write(outputName);
    }
```
the size is 41.4 KB (42,441 bytes)

I just opened it and saved. The size is significantly reduced. 
How to keep the size?

The file is attached:
![GIF_FIle](https://github.com/dlemstra/Magick.NET/assets/22297424/022fb20e-a200-4a9a-b966-da5077e92a73)

**Comment by dlemstra** [comment]
For a gif you should use `MagickImageCollection` instead. Now you are only reading the first frame of the image and not all frames.

**Reply by VadymIan** [reply]
Cool! Thank you for helping me with *.gif files. 
One more question: how to properly handle other formats without size loses?

```
using (var image = new MagickImage(imageStream))
    {
            image.Write(outputName);
    }
```

using this snippet of code leads me to loses of file size without manipulating

jpg: 166 KB (170,604 bytes) -> 85.6 KB (87,659 bytes)
png: 2.88 MB (3,027,164 bytes) -> 1.74 MB (1,834,172 bytes)

I was trying to save the initial settings to the resulted file, but it has no effect:
```
image.Settings.Compression = image.Compression;
image.Settings.Depth = image.Depth;
image.Settings.BackgroundColor = image.BackgroundColor;
image.Settings.BorderColor = image.BorderColor;
image.Settings.ColorSpace = image.ColorSpace;
image.Settings.ColorType = image.ColorType;
image.Settings.Density = image.Density;
image.Settings.Endian = image.Endian;
```

Probably I'm doing something wrong, could you please point me to the right direction. What should I pay attention to?

![image20](https://github.com/dlemstra/Magick.NET/assets/22297424/fbb15ef9-24f4-48e9-b3de-61be77fafbef)

**Reply by dlemstra** [reply]
The size of the file is reduced because you are removing meta data and compression the pixels again. If you don't want that you might need to search for a different tool that can be used to only remove the metadata.

**Reply by VadymIan** [reply]
Please pay attention that I don't remove metadata and apply compession:
```
using (var image = new MagickImage(imageStream))
    {
            image.Write(outputName);
    }
```
I'm saing that the file size is reduced when I pass stream to `MagickImage` class and call `Write` method which supposed I believe to recreate image with the same settings and size, but this doesn't happen because file size is significantly reduced.
The question is: Is it expected behavour, bug or some steps are missed to save the initial file size?

Thank you!

**Reply by dlemstra** [reply]
This is expected behavior. You are re-compressing the image so that could result in a smaller or bigger image.

**Reply by VadymIan** [reply]
okay, could please advice me how to keep the initial size after opening file via `magickImage` and writing to output file?

**Reply by dlemstra** [reply, marked as answer]
As I said in my earlier comment this is expected behavior. When the image is being read the pixels are decompressed and when the image is being written those pixels will be compressed again. This means that the file can be either smaller or larger. That will always happen with this library and cannot be avoided.

**Reply by VadymIan** [reply]
Thank you for explanation!
